### ISSUE TITLE
ENH: finite support discretized counts, unbalanced ordinal model

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#7784]
random idea while thinking about beta-binomial #2632

designer/flexible distributions for counts with finite support.

~~2~~ 3 possible models:

- DiscretizedCount with observation specific upper bound on the (discretized) distribution, latent variable is independent of upper bound.
  maybe with different discretization schemes
  - dropout: same cutoff for all obs, but truncated when individual (exogenously) drops out.
  - cutoffs depend on n_trials: e.g. finite support of latent variable distribution and it is split evenly across opportunities
    (I'm not sure how that would work. Increments for discretization would likely change)
- unbalanced ordinal model (not all observations have the same number of choices, ordinal extension of Binomial with unequal n_trials), 
   - example: questionaire differ in nested coarseness, known splits of categories. some get 5 scale, some get 3 scale, the 3scale is a coarsening of the 5 scale partition of the latent variable space, e.g. (0,1) -> 0, then 1st cutoff of coarses scale corresponds to 2nd cutoff of finer scale.
   - example: only 2 types of questionaires, unrestricted cutoffs. Then, we would have to estimate separate cutoffs for each questionaire type, but a different number of cutoffs for each. (This might be feasible to implement)
- maybe we can just use a truncated version of an existing discretized count or ordinal distribution, (that should corresponds to the "dropout" version)
 
(example: political spectrum, continuous support on [-3, 3], answer is for discretized scale, but coarseness of scales differ)
(example: movie, book,,, ratings: merge ratings based on 3, 4, and 5 star scales, and numeric out of 100)
(example: grades for exams on different scales)

Related: Conditional Logit in unbalanced panel where we condition on the total number of units of a panel unit.

Maybe also related to survey questions where respondents can mark multiple answer. (Of the following 10 answers, mark all that apply to you. Maybe not for the answer, but for the count of how many marks a respondent made.)

and to get more distracted:
endogenous coarsening by topic experts
example rate out of 100.
respondents answer in 10 point intervals if they don't know the topic
respondents answer on a fine scale if they know the topic

(How many words for snow do eskimos/inuit have? https://www.k-international.com/blog/which-language-has-the-most-words-for-snow/ )

**Comment by josef-pkt** [comment]
another flexible two parameter distribution for finite count  (counts in a fixed finite interval)

Janowski, Lucjan, Bogdan Ćmiel, Krzysztof Rusek, Jakub Nawała, and Zhi Li. 2019. “Generalized Score Distribution.” ArXiv:1909.04369 [Cs, Stat], September. http://arxiv.org/abs/1909.04369.

allows underdispersion compared to Binomial.
Beta-Binomial has smallest variance equal to Binomial

(I downloaded it a year ago, but hadn't looked at it)

web access to code https://colab.research.google.com/drive/1ioM4JqUaEA8fHJH9V-0iphHt4C9zmI0i
(linked to in article)
and https://github.com/LucjanJanowski/Generalized-Score-Distribution/blob/master/Generalized_Score_Distribution.ipynb
explicit MIT license, code uses tensorflow for optimization and derivatives
